Source/Core/DolphinWX/VideoConfigDiag.cpp

```diff
@@ -751,24 +742,38 @@ void VideoConfigDiag::PopulatePostProcessingShaders()
 	button_config_pp->Enable(postprocessing_shader.HasOptions());
 }
 
-void VideoConfigDiag::OnSSAAClick(wxCommandEvent& event)
+void VideoConfigDiag::PopulateAAList()
 {
-	// Check the checkbox status and not the config option because config hasn't changed yet.
-	vconfig.bSSAA = ssaa_checkbox->IsChecked();
-	RefreshAAList();
-}
-
-void VideoConfigDiag::RefreshAAList()
-{
-	choice_aamode->Clear();
-	const std::string& suffix = vconfig.bSSAA ? "x SSAA" : "x MSAA";
-
 	for (int mode : vconfig.backend_info.AAModes)
 	{
 		if (mode == 1)
+		{
 			choice_aamode->AppendString(_("None"));
+		}
 		else
-			choice_aamode->AppendString(std::to_string(mode) + suffix);
+		{
+			const std::string mode_string = std::to_string(mode);
+			choice_aamode->AppendString(mode_string + "x MSAA");
+			choice_aamode->AppendString(mode_string + "x SSAA");
+		}
+	}
+
+	int selected_mode = vconfig.iMultisampleMode;
```

Comment:
I don't know how it should be handled in the backend and config. Are you asking about that, or are you asking about problems that might occur with `choice_aamode->SetSelection(selected_mode)`?